specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/kustoPool.json

```diff
@@ -51,6 +51,90 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Synapse/skus": {
+      "get": {
+        "description": "Lists eligible SKUs for Kusto Pool resource.",
+        "x-ms-examples": {
+          "KustoPoolsListSkus": {
+            "$ref": "./examples/KustoPoolsListSkus.json"
+          }
+        },
+        "operationId": "KustoPool_ListSkus",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/SkuDescriptionList"
+            }
+          },
+          "default": {
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            },
+            "description": "Error response describing why the operation failed."
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Synapse/locations/{location}/kustoPoolCheckNameAvailability": {
```

Comment:
why can't the existing checkNameAvailability API be used?  It takes resource type as a parameter.  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#check-name-availability-requests

Reply:
I see this just moved up in the spec and already existed so it can't be removed anyway.  Something to note for future top level name availability requests though

Reply:
Thanks for the heads up!
I should also note that the KustoPool resource is derived from the Kusto cluster resource (kusto.json) as part of the Kusto swagger, so there might be logic inheritance that comes straight from the original Kusto swagger.